### PR TITLE
Initial Okta objects.

### PR DESCRIPTION
**Pull request by mdwn**
To support the Okta integration, the initial Okta objects have been added.

Note: The associated RFD can be found [here](https://github.com/gravitational/teleport.e/blob/master/rfd/0003e-application-access-okta-integration.md).

### CODE REVIEW

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -5043,3 +5043,111 @@ message UserGroupV1 {
...
+  // Target is the target of the assignment action.
+  OktaAssignmentActionTargetV1 target = 2 [(gogoproto.jsontag) = "target"];
+}
+
+// OktaAssignmentActionTargetV1 is the target of an Okta assignment action.
+message OktaAssignmentActionTargetV1 {
+  // OktaAssignmentActionTargetType is the type of Okta object that an action is targeting.
+  enum OktaAssignmentActionTargetType {
+    // APPLICATION indicates the target is an application.
+    APPLICATION = 0;
```
Comment (ravicious) [inline api/proto/teleport/legacy/types/types.proto:5144]: Same as above regarding "application" being the default.

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -5043,3 +5043,111 @@ message UserGroupV1 {
...
+  // Actions is a list of Okta actions to take on a user.
+  repeated OktaAssignmentActionV1 Actions = 2 [(gogoproto.jsontag) = "actions"];
+}
+
+// OktaAssignmentActionV1 is a Okta assignment action.
+message OktaAssignmentActionV1 {
+  // OktaAssignmentStatus represents the status of an action in an Okta assignment.
+  enum OktaAssignmentActionStatus {
+    // PENDING indicates the action has not yet been applied.
+    PENDING = 0;
```
[the remark this answers is not in the document]
Comment (mdwn) [reply api/proto/teleport/legacy/types/types.proto:5122]: Hrm, good eye here. That seems like reasonable thing to have here. I'll make this change for this and the other you brought up.

**Review by ravicious — APPROVED** [review]

File: api/types/okta.go [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+	case OktaAssignmentActionV1_SUCCESSFUL:
+		return constants.OktaAssignmentActionStatusSuccessful
+	case OktaAssignmentActionV1_FAILED:
+		return constants.OktaAssignmentActionStatusFailed
+	case OktaAssignmentActionV1_CLEANED_UP:
+		return constants.OktaAssignmentActionStatusCleanedUp
+	case OktaAssignmentActionV1_CLEANUP_FAILED:
+		return constants.OktaAssignmentActionStatusCleanupFailed
+	default:
+		return constants.OktaAssignmentActionStatusPending
```
Comment (tobiaszheller) [inline api/types/okta.go:270]: unkown should be mapped to pending?

File: api/types/okta.go [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+// GetTargetType returns the target type of the action.
+func (o *OktaAssignmentActionV1) GetTargetType() string {
+	switch o.Target.Type {
+	case OktaAssignmentActionTargetV1_APPLICATION:
+		return constants.OktaAssignmentActionTargetApplication
+	case OktaAssignmentActionTargetV1_GROUP:
+		return constants.OktaAssignmentActionTargetGroup
+	default:
+		return constants.OktaAssignmentActionTargetApplication
```
Comment (tobiaszheller) [inline api/types/okta.go:282]: same question here, i guess whole point of adding uknown into proto enum is to handle it later

**Review by tobiaszheller — COMMENTED** [review]
left 2 questions, otherwise lgtm

File: api/types/okta.go (hunk 3 above)
Comment (mdwn) [reply api/types/okta.go:270]: Ahhhh you're probably right that it should be mapped to unknown, I didn't update these after adding that status. I'll add unknown to these.

**Review by tobiaszheller — APPROVED** [review]

**Pull request merged**